orbit/models/ktrlite.py

```diff
@@ -0,0 +1,584 @@
+        Returns
+        -------
+        pd.DataFrame
+            data with computed fourier series attached
+        """
+        if len(self._seasonality) > 0:
+            for idx, s in enumerate(self._seasonality):
+                order = self._seasonality_fs_order[idx]
+                df, _ = make_fourier_series_df(df, s, order=order, prefix='seas{}_'.format(s), shift=shift)
+
+        return df
+
+    def _set_valid_response_attributes(self):
+        if self._seasonality:
+            max_seasonality = np.round(np.max(self._seasonality)).astype(int)
+            self._response_mean = np.nanmean(self._response[:max_seasonality])
+        else:
+            self._response_mean = np.nanmean(self._response)
+
+        self._is_valid_response = ~np.isnan(self._response)
+        # [0] to convert tuple back to array
+        self._which_valid_response = np.where(self._is_valid_response)[0]
+        self._num_of_valid_response = len(self._which_valid_response)
+        self._response_sd = np.nanstd(self._response)
+
+    def _set_regressor_matrix(self, df):
+        # init of regression matrix depends on length of response vector
+        self._regressor_matrix = np.zeros((self._num_of_observations, 0), dtype=np.double)
+        if self._num_of_regressors > 0:
+            self._regressor_matrix = df.filter(items=self._regressor_col,).values
+
+    @staticmethod
+    def get_gap_between_dates(start_date, end_date, freq):
+        diff = end_date - start_date
+        gap = np.array(diff / np.timedelta64(1, freq))
+
+        return gap
+
+    @staticmethod
+    def set_knots_tp(knots_distance, cutoff):
+        # start in the middle
+        knots_idx_start = round(knots_distance / 2)
+        knots_idx = np.arange(knots_idx_start, cutoff, knots_distance)
+
+        return knots_idx
+
+    def _set_kernel_matrix(self, df):
+        # Note that our tp starts by 1; to convert back to index of array, reduce it by 1
+        tp = np.arange(1, self._num_of_observations + 1) / self._num_of_observations
+
+        # this approach put knots in full range
+        self._cutoff = self._num_of_observations
+
+        # kernel of level calculations
+        if self._level_knot_dates is None:
+            if self.level_knot_length is not None:
+                # TODO: approximation; can consider level_knot_length directly as step size
+                knots_distance = self.level_knot_length
+            else:
+                number_of_knots = round(1 / self.span_level)
+                knots_distance = math.ceil(self._cutoff / number_of_knots)
+
+            knots_idx_level = self.set_knots_tp(knots_distance, self._cutoff)
+            self._knots_idx_level = knots_idx_level
+            self._knots_tp_level = (1 + knots_idx_level) / self._num_of_observations
+            self._level_knot_dates = df[self.date_col].values[knots_idx_level]
+        else:
+            # FIXME: this only works up to daily series (not working on hourly series)
```

Comment:
did we fix this already?

Reply:
yea, it is fixed. this line of comment is removed in the PR then